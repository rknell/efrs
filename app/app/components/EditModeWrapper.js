import React, { Component } from 'react'
import { StyleSheet, TouchableOpacity, View, TouchableNativeFeedback, Platform } from 'react-native'
import IconFA from 'react-native-vector-icons/FontAwesome'
import GlobalStyles from '../services/GlobalStyle'

let UsedTouchable

class AndroidTouchable extends Component {
  render () {
    return (<TouchableNativeFeedback onPress={this.props.onPress}>
      <View style={this.props.style}>{this.props.children}</View>
    </TouchableNativeFeedback>)
  }
}

export default class EditModeWrapper extends Component {
  constructor (props) {
    super(props)
    if (Platform.OS === 'android') {
      UsedTouchable = AndroidTouchable
    } else {
      UsedTouchable = TouchableOpacity
    }
  }

  render () {
    if (this.props.editMode) {
      return (<View style={{ flexDirection: 'row' }}>
        <View style={{ flexDirection: 'row', alignItems: 'center', borderBottomWidth: 1, borderBottomColor: 'rgba(0,0,0,0.1)', backgroundColor: GlobalStyles.style.primaryColour }}>
          <UsedTouchable onPress={() => this.props.onMoveUp()} style={{ padding: 5, paddingLeft: 10 }}><IconFA name='chevron-up' size={20} style={{ color: GlobalStyles.style.primaryColourText }} /></UsedTouchable>
          <UsedTouchable onPress={() => this.props.onMoveDown()} style={{ padding: 5, paddingRight: 15 }}><IconFA name='chevron-down' size={20} style={{ color: GlobalStyles.style.primaryColourText }} /></UsedTouchable>
        </View>
        <UsedTouchable onPress={this.props.onPressEdit} style={{ flex: 1 }}>{this.props.children}</UsedTouchable>
        <View style={styles.borderBottom} />
      </View>)
    } else {
      return this.props.children
    }
  }
}

const styles = StyleSheet.create({
  borderBottom: {
    flexDirection: 'row',
    alignItems: 'center',
    borderBottomWidth: 1,
    borderBottomColor: 'rgba(0,0,0,0.1)'
  },
  editWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
    borderBottomWidth: 1,
    borderBottomColor: 'rgba(0,0,0,0.1)'
  }
})
