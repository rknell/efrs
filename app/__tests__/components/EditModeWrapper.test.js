import React from 'react'
import EditModeWrapper from '../../app/components/EditModeWrapper'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { Text } from 'react-native'
import IconFA from 'react-native-vector-icons/FontAwesome'

it('should render only children in edit mode', () => {
  const wrapper = shallow(<EditModeWrapper><Text>TEST</Text></EditModeWrapper>)
  expect(toJSON(wrapper)).toMatchSnapshot()
  expect(wrapper.find(IconFA).length).toBe(0)
})

it('should render in edit mode', () => {
  const wrapper = shallow(<EditModeWrapper editMode />)
  expect(toJSON(wrapper)).toMatchSnapshot()
  expect(wrapper.find(IconFA).length).toBe(2)
})
