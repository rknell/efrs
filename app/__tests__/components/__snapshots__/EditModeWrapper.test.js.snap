// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render in edit mode 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "dodgerblue",
        "borderBottomColor": "rgba(0,0,0,0.1)",
        "borderBottomWidth": 1,
        "flexDirection": "row",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "padding": 5,
          "paddingLeft": 10,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        name="chevron-up"
        size={20}
        style={
          Object {
            "color": "rgb(255, 255, 255)",
          }
        }
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "padding": 5,
          "paddingRight": 15,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        name="chevron-down"
        size={20}
        style={
          Object {
            "color": "rgb(255, 255, 255)",
          }
        }
      />
    </TouchableOpacity>
  </View>
  <TouchableOpacity
    activeOpacity={0.2}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "rgba(0,0,0,0.1)",
        "borderBottomWidth": 1,
        "flexDirection": "row",
      }
    }
  />
</View>
`;

exports[`should render only children in edit mode 1`] = `
<Text>
  TEST
</Text>
`;
